components/Promotion/interfaces.rst

```diff
@@ -0,0 +1,199 @@
+
+.. _Sylius API PromotionSubjectInterface: http://api.sylius.org/Sylius/Component/Promotion/Model/PromotionSubjectInterface.html
+
+
+.. _component_promotion_model_promotion-countable-subject-interface:
+
+PromotionCountableSubjectInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To be able to count the object's promotion subjects, the object class needs to implement
+the ``PromotionCountableSubjectInterface``.
+
+.. note::
+
+    This interface extends the :ref:`component_promotion_model_promotion-subject-interface`. |br|
+    You will find more information about this interface in `Sylius API PromotionCountableSubjectInterface`_.
+
+.. _Sylius API PromotionCountableSubjectInterface: http://api.sylius.org/Sylius/Component/Promotion/Model/PromotionCountableSubjectInterface.html
+
+
+.. _component_promotion_model_promotion-coupon-aware-subject-interface:
+
+PromotionCouponAwareSubjectInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To make the object able to get its associated coupon, the object class needs to implement
+the ``PromotionCouponAwareSubjectInterface``.
+
+.. note::
+
+    This interface extends the :ref:`component_promotion_model_promotion-subject-interface`. |br|
+    You will find more information about this interface in `Sylius API PromotionCouponAwareSubjectInterface`_.
+
+.. _Sylius API PromotionCouponAwareSubjectInterface: http://api.sylius.org/Sylius/Component/Promotion/Model/PromotionCouponAwareSubjectInterface.html
+
+
+.. _component_promotion_model_promotion-coupons-aware-subject-interface:
+
+PromotionCouponsAwareSubjectInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To make the object able to get its associated coupons collection, the object class needs to implement
+the ``PromotionCouponsAwareSubjectInterface``.
+
+.. note::
+
+    This interface extends the :ref:`component_promotion_model_promotion-subject-interface`. |br|
+    You will find more information about this interface in `Sylius API PromotionCouponsAwareSubjectInterface`_.
+
+.. _Sylius API PromotionCouponsAwareSubjectInterface: http://api.sylius.org/Sylius/Component/Promotion/Model/PromotionCouponsAwareSubjectInterface.html
+
+
+Services Interfaces
+-------------------
+
+.. _component_promotion_model_action-action-interface:
+
+PromotionActionInterface
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by models that do actions on promotions. These are the models that execute actions on the promotion subjects.
+
+.. note::
+
+    You will find more information about this interface in `Sylius API PromotionActionInterface`_.
+
+.. _Sylius API PromotionActionInterface: http://api.sylius.org/Sylius/Component/Promotion/Action/PromotionActionInterface.html
+
+
+.. _component_promotion_checker_promotion-eligibility-checker-interface:
+
+PromotionEligibilityCheckerInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to check the promotions' eligibility, the object class has to implement the ``PromotionEligibilityCheckerInterface``.
```

Comment:
This is not very clear. I'd say that ``Object responsible for checking the eligibility of a promotion subject should implement this interface``. Current version made me think that I need to implement it on the subject itself.